tests/ui/stacked_ifs.rs

```diff
@@ -0,0 +1,92 @@
+#![warn(clippy::stacked_ifs)]
+#![allow(clippy::manual_unwrap_or_default)]
+
+macro_rules! m {
+    () => {
+        let a = true;
+        let b = false;
+        if if a { b } else { true } {
+            println!("True");
+        }
+    };
+}
```

Comment:
This is good to test the first `from_expansion` check, it would be good to have another that tests the `from_expansion` check of the condition like https://github.com/rust-lang/rust-clippy/pull/12548#discussion_r1536845445